### ISSUE TITLE
update CIs scripts to skip execution only after checking all commits changes

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#7023]
identified at https://github.com/checkstyle/checkstyle/issues/2116#issuecomment-524727108

most CIs have code like - https://github.com/checkstyle/checkstyle/blob/master/.travis.yml#L285
it is was created to skip other CIs is there is change in configuration of only particular CI, it is was good 
but it worked when we have single commit PR, that is 99% cases in our workflow.

But there was a case in referenced incident.

TODO:
review all CIs on how they do clone and if that is possible to get amount of commits in PR, if yes...get diff of files in commits between master and branch head (PRs).

**Comment by rnveach** [comment]
Fix was merged, testing fix in another CI.

**Comment by romani** [comment]
I presume we finished this issue, is something left, lets open new issue
